docs/admin-manual/file-and-cred-transfer.rst

```diff
@@ -524,186 +524,181 @@ htvault-config
-   (``.well-known/openid-configuration``) to the public key file and the key id to
-   use when signing tokens. The key id is the value of the "kid" property in the
-   public key JWKS file. For example, if the "kid" is "abc0":
-   
-   .. code-block:: condor-config
-   
-       LOCAL_CREDMON_ISSUER = https://example.com/scitokens
-       LOCAL_CREDMON_KEY_ID = abc0
-   
-   Finally, set the lifetime and scopes of the tokens. A templating system is
-   available for setting scopes based on the submitter's system username.
-   Optionally, if ``LOCAL_CREDMON_AUTHZ_GROUP_TEMPLATE`` and
-   ``LOCAL_CREDMON_AUTHZ_GROUP_MAPFILE`` are set, a mapfile can be used
-   to append additional scopes based on all of the values that the submitter's
-   system username maps to, which is typically useful for group-accessed
-   locations.
-   
-   .. code-block:: condor-config
-   
-       LOCAL_CREDMON_TOKEN_LIFETIME = 1200
-       LOCAL_CREDMON_AUTHZ_TEMPLATE = read:/user/{username} write:/user/{username}
-       LOCAL_CREDMON_AUTHZ_GROUP_TEMPLATE = read:/groups/{groupname} write:/groups/{groupname}
-       LOCAL_CREDMON_AUTHZ_GROUP_MAPFILE = /etc/condor/local_credmon_group_map
-   
-   For example, supposed that user "bob" should have access to
-   ``/groups/projectA`` and ``/group/projectB`` and "alice" should have access to
-   ``/groups/projectB``, the mapfile (``/etc/condor/local_credmon_group_map``)
-   might look like:
-   
-   .. code-block:: text
-   
-       * bob projectA,projectB
-       * alice projectB
-   
-   Configuring HTCondor to automatically create SciTokens for jobs
-   '''''''''''''''''''''''''''''''''''''''''''''''''''''''''''''''
-   
-   At this point, the local issuer is configured to be able to generate valid
-   SciTokens. A final, optional step is to install a job transform that tells
-   HTCondor to automatically create tokens and send them along with every
-   submitted job.
-   The following example is such a job transform that will do this for all
-   vanilla, container, and local universe jobs:
-   
-   .. code-block:: condor-config
-   
-       JOB_TRANSFORM_AddSciToken @=end
-       [
-           Requirements = (JobUniverse == 5 || JobUniverse == 12);
-           Eval_Set_OAuthServicesNeeded = strcat( "scitokens ", OAuthServicesNeeded ?: "");
-       ]
-       @end
-       JOB_TRANSFORM_NAMES = $(JOB_TRANSFORM_NAMES) AddSciToken
-   
-   This example also assumes that ``LOCAL_CREDMON_PROVIDER_NAME = scitokens``,
-   replace ``"scitokens "`` in the ``strcat`` function to match this name if
-   different.
+Automatic Issuance of SciTokens Credentials
+-------------------------------------------
+
+The ``condor-credmon-local`` rpm package includes a SciTokens "local
+issuer."  Once enabled, no web browser authorization is needed for users
+to be issued a SciToken when submitting a job. The claims of the SciToken
+are entirely controlled by the HTCondor configuration (as read by the
```

Comment:
Shouldn't following line refer to the condor_credmon_local daemon?

Reply:
Right now there is no `condor_credmon_local`, the local issuer credmon still runs inside the `CREDMON_OAUTH = condor_credmon_oauth` daemon. When first trying to tackle this, I started to split the credmons into two before coming to the realization that this would require all sorts of changes to the `condor_credd` and/or `condor_submit` code to look for a `CREDMON_LOCAL` daemon and its associated config knobs. So for now, I instead found that the `condor_credmon_oauth` code as written works fine in a local-only mode without making other code changes, so took a minimal packaging-only changes approach.

Reply:
OK